x-pack/legacy/plugins/apm/public/components/app/TraceLink/index.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import url from 'url';
+import styled from 'styled-components';
+import { Redirect } from 'react-router-dom';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { QueryParams } from '../../shared/Links/apm/ExternalLinks';
+
+const CentralizedContainer = styled.div`
+  height: 100%;
+  display: flex;
+`;
+
+const redirectToTransactionDetailPage = ({
+  transaction,
+  rangeFrom,
+  rangeTo
+}: { transaction: Transaction } & Partial<QueryParams>) => {
+  const dateRange = rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {};
+  return url.format({
+    pathname: `/services/${transaction.service.name}/transactions/view`,
+    query: {
+      traceId: transaction.trace.id,
+      transactionId: transaction.transaction.id,
+      transactionName: transaction.transaction.name,
+      transactionType: transaction.transaction.type,
+      ...dateRange
+    }
+  });
+};
+
+const redirectToTracePage = (traceId: string) =>
+  url.format({
+    pathname: `/traces`,
+    query: {
+      kuery: encodeURIComponent(`trace.id : "${traceId}"`)
```

Comment:
```suggestion
      kuery: encodeURIComponent(`${TRACE_ID} : "${traceId}"`)
```

Reply:
Btw. now you are add it, can you also fix this to use constants?

https://github.com/elastic/kibana/blob/d1de029b8873e29c35df4bf6ce95f987efd83874/x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/WaterfallItem.tsx#L214-L216